### PR TITLE
[Enhancement][BugFix] fix spillable operator's name in profile

### PR DESCRIPTION
**Pull request by silverbullet233**
## Problem Summary:
Fixes # (issue)

1. fix some spillable operator's names so that our web ui can render the query profile
2. add some session variables to control whether each type of operator can spill, which can facilitate us to troubleshoot the implementation of spill operators in complex queries

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 8 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|4|8|50.00%|[2154, 2155, 2156, 2157]|

**Review by satanson — DISMISSED** [review]

**Review by stdpain — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 10 (40.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|4|10|40.00%|[2155, 2156, 2157, 2158, 2159, 2160]|

**Review by satanson — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -704,6 +709,18 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
...
+    private boolean enableHashJoinSpill = true;
+
+    @VariableMgr.VarAttr(name = ENABLE_AGG_SPILL, flag = VariableMgr.INVISIBLE)
+    private boolean enableAggSpill = true;
+
+    @VariableMgr.VarAttr(name = ENABLE_AGG_DISTINCT_SPILL, flag = VariableMgr.INVISIBLE)
+    private boolean enableAggDistinctSpill = true;
+
+    @VariableMgr.VarAttr(name = ENABLE_SORT_SPILL, flag = VariableMgr.INVISIBLE)
+    private boolean enableSortSpill = true;
```
Comment (fzhedu) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:722]: how about union them into a mask? per bit stands for different operators.

Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:722]: > how about union them into a mask? per bit stands for different operators.

I think this is not intuitive enough for users, so I tend to keep the current way and just use mask on thrift

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 2]
```diff
@@ -704,6 +709,18 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     @VariableMgr.VarAttr(name = ENABLE_SPILL)
     private boolean enableSpill = false;
 
+    @VariableMgr.VarAttr(name = ENABLE_HASH_JOIN_SPILL, flag = VariableMgr.INVISIBLE)
+    private boolean enableHashJoinSpill = true;
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:713]: I don't think we need so many session variables.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[2153]|

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 3]
```diff
@@ -704,6 +705,20 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
...
+    // it uses a bit to identify whether the spill of each operator is in effect, 0 means no, 1 means yes
+    // at present, only the lowest 4 bits are meaningful, corresponding to the four operators
+    // HASH_JOIN, AGG, AGG_DISTINCT and SORT respectively (see TSpillableOperatorType in InternalService.thrift)
+    // e.g.
+    // if spillable_operator_mask & 1 != 0, hash join operator can spill
+    // if spillable_operator_mask & 2 != 0, agg operator can spill
+    // if spillable_operator_mask & 4 != 0, agg distinct operator can spill
+    // if spillable_operator_mask & 8 != 0, sort operator can spill
+    // ...
+    // default value is 15, means all operators can spill
```
Comment (fzhedu) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:718]: prefer `-1` to avoid upgrading issues when it supports more operators.

**Review by satanson — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 3 above)
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java]: > prefer `-1` to avoid upgrading issues when it supports more operators.

fixed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[2153]|

**Review by kangkaisen — APPROVED** [review]

**Review by fzhedu — APPROVED** [review]
LGTM

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**